pkg/reconciler/buildtemplate/buildtemplate.go

```diff
@@ -0,0 +1,118 @@
+	listers "github.com/knative/build/pkg/client/listers/build/v1alpha1"
+)
+
+const controllerAgentName = "buildtemplate-controller"
+
+// Reconciler is the controller.Reconciler implementation for BuildTemplates resources
+type Reconciler struct {
+	// kubeclientset is a standard kubernetes clientset
+	kubeclientset kubernetes.Interface
+	// buildclientset is a clientset for our own API group
+	buildclientset clientset.Interface
+
+	buildTemplatesLister listers.BuildTemplateLister
+
+	// Sugared logger is easier to use but is not as performant as the
+	// raw logger. In performance critical paths, call logger.Desugar()
+	// and use the returned raw logger instead. In addition to the
+	// performance benefits, raw logger also preserves type-safety at
+	// the expense of slightly greater verbosity.
+	Logger *zap.SugaredLogger
+}
+
+// Check that we implement the controller.Reconciler interface.
+var _ controller.Reconciler = (*Reconciler)(nil)
+
+func init() {
+	// Add buildtemplate-controller types to the default Kubernetes Scheme so Events can be
+	// logged for buildtemplate-controller types.
+	buildscheme.AddToScheme(scheme.Scheme)
+}
+
+// NewController returns a new build template controller
+func NewController(
+	logger *zap.SugaredLogger,
+	kubeclientset kubernetes.Interface,
+	buildclientset clientset.Interface,
+	buildTemplateInformer informers.BuildTemplateInformer,
+) *controller.Impl {
+
+	// Enrich the logs with controller name
+	logger = logger.Named(controllerAgentName).With(zap.String(logkey.ControllerType, controllerAgentName))
+
+	r := &Reconciler{
+		kubeclientset:        kubeclientset,
+		buildclientset:       buildclientset,
+		buildTemplatesLister: buildTemplateInformer.Lister(),
+		Logger:               logger,
+	}
+	impl := controller.NewImpl(r, logger, "BuildTemplates")
+
+	logger.Info("Setting up event handlers")
+	// Set up an event handler for when BuildTemplate resources change
+	buildTemplateInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc:    impl.Enqueue,
+		UpdateFunc: controller.PassNew(impl.Enqueue),
+	})
+
+	return impl
+}
+
+// Reconcile implements controller.Reconciler
+func (c *Reconciler) Reconcile(ctx context.Context, key string) error {
+	logger := logging.FromContext(ctx)
+
+	// Convert the namespace/name string into a distinct namespace and name
+	namespace, name, err := cache.SplitMetaNamespaceKey(key)
+	if err != nil {
+		logger.Errorf("invalid resource key: %s", key)
+		return nil
+	}
+
+	// Get the BuildTemplate resource with this namespace/name
+	_, err = c.buildTemplatesLister.BuildTemplates(namespace).Get(name)
+	if errors.IsNotFound(err) {
+		// The BuildTemplate resource may no longer exist, in which case we stop processing.
+		logger.Errorf("buildtemplate %q in work queue no longer exists", key)
+		return nil
+	} else if err != nil {
+		return err
+	}
+
+	// TODO: Meaningful reconciliation.
```

Comment:
What does this mean in this context? Is there any meaningful reconciliation to do for build templates? (same with `ClusterBuildTemplate`s)